app/javascript/segment_graph_builder.js

```diff
@@ -0,0 +1,73 @@
+import {build_segment_graph} from "graphs/segment_duration_graph.js"
+import {Spinner} from 'spin.js'
+
+$(function() {
+  const graphHolders = document.getElementsByClassName('segment-graph-holder')
+  if(graphHolders.length === 0) {
+    return
+  }
+
+  let spinners = []
+
+  for(const graphHolder of graphHolders) {
+    const spinner = new Spinner({
+      lines: 3,
+      length: 15,
+      width: 1,
+      radius: 0,
+      corners: 1,
+      rotate: 90,
+      direction: 1,
+      color: '#FFFFFF',
+      speed: 0.5,
+      trail: 30,
+      shadow: false,
+      hwaccel: true,
+      position: 'relative'
+    })
+    spinner.spin(graphHolder)
+    spinners.push(spinner)
+  }
+
+  const runJSON = fetch(`/api/v4/runs/${gon.run.id}?historic=1`, {
+    headers: {
+      accept: 'application/json'
+    }
+  }).then(function(response) {
+    if (response.ok) {
+      return response.json()
+    }
+    throw new Error('Request for run from api failed')
+  })
+    
+  runJSON.then(function(json) {
+    spinners.forEach(function(spinner) {
+      spinner.stop()
+    })
+  }).catch(function(error) {
+    spinners.forEach(function(spinner) {
+      spinner.stop()
+    })
+    for(const graphHolder of document.getElementsByClassName('graph-holder')) {
+      graphHolder.find('.panel-title').find('h1').text('Error retrieving data for graphs')
+    }
+  })
+
+  if (document.getElementById('run-timer').dataset.timer === 'livesplit') {
```

Comment:
Is there a reason to not use gon for this as it is used above to get the run id?

Reply:
Hmm, I was originally thinking to migrate off gon to keep us sane about passing variables through like this (so we don't do the accidentally-load-every-segment-history thing), but now that I think about it, it's just as easy to accidentally pass too much through this way also, so there's really no benefit. We will just have to be careful in PRs about catching that kind of stuff.